tests/integration/test_core.py

```diff
@@ -540,3 +545,111 @@ async def test_openid_plugin(
         assert (
             "administrator" in wordpress_client.list_roles()
         ), "An launchpad OpenID account should be associated with the WordPress admin user"
+
+
+async def test_prometheus_integration(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name: str,
+    unit_ip_list: list[str],
+):
+    """
+    arrange: after WordPress charm has been deployed and relations established.
+    act: prometheus charm joins relation
+    assert: prometheus metrics endpoint for prometheus is active and prometheus has active scrape
+        targets.
+    """
+    assert ops_test.model
+    prometheus: Application = await ops_test.model.deploy(
+        "prometheus-k8s", channel="stable", trust=True
+    )
+
+    await ops_test.model.relate(application_name, prometheus.name)
+    await ops_test.model.wait_for_idle(apps=[application_name, prometheus.name], status="active")
```

Comment:
I'm not sure, does `apps` have to be a list? Or could it also be a tuple given it is a constant?

Reply:
Similar comment for any other constant collections

Reply:
It does seem like the `juju.Model` strictly requires it to be a list[str]
```
# from juju/model.py:L2626
if apps is not None and (not isinstance(apps, list) or
                                 any(not isinstance(o, str)
                                     for o in apps)):
            raise JujuError(f'Expected a List[str] for apps, given {apps}')
```